builtin/core.wren

```diff
@@ -21,6 +21,16 @@ class Sequence {
     }
     return true
   }
+
+  reduce(acc, f) {
+    for (element in this) {
+      acc = f.call(acc, element)
+    }
+    return acc
+  }
+
+  reduce(f) { this[1..-1].reduce(this[0], f) }
```

Comment:
>  we should swap the order of arguments in the 2-arity reduce.

This is an interesting style point. In other dynamic languages that don't support overloading, I think there's a strong convention to have extra arguments only be added to the end. That makes it easier to describe and name the earlier parameters since they have a consistent interpretation in the different simulated overloads.

My experience from statically typed languages that actually support overloading is that it's less important there. Of course, you don't want to just shuffle parameters arbitrarily, but I think you can get away with things like:

```
random(max) { ... }
random(min, max) { ... }
```

I think Wren is closer to a statically-typed language here since we do have real overloads by arity. My feeling is that the accumulator seed naturally comes before the function in `reduce`. Putting the function last *certainly* makes it play better with block args. Actually, now that I think about it, that alone is enough justification to me to have the fn at the end in both arities.